include/flamegpu/io/factory.h

```diff
@@ -36,11 +36,16 @@ inline std::string getFileExt(const std::string& s) {
 */
 class ReaderFactory {
  public:
-    static StateReader *createReader(const std::string &model_name, const std::unordered_map<std::string, std::shared_ptr<AgentPopulation>> &model_state, const std::string &input) {
+    static StateReader *createReader(
+        const std::string &model_name,
+        const unsigned int &instance_id,
```

Comment:
Is `instance_id` the instance ID of the StateReader, or of the Simulation?

This probably applies to every use of `instance_id`.

Reply:
Always `Simulation` instance, this can be resolved with comments. I updated all the comments that already existed. But some places didn't have comments to begin with.

I'll add comments to the IO classes.

Reply:
(It's required because that's used to access the Environment from within the IO method.)

Reply:
I wasn't sure if Model name was strictly necessary there, so I left it in.

Reply:
would prefer it to be `simulation_instance_id` in that case tbh, Comments work so fine without, just not keen on the ambiguity of it.

Reply:
I've updated IO to `sim_instance_id` throughout, i'm less concerned about the rest. Just waiting for tests to build incase i broke something with refactor tools.